File: plotting.py
"""
This file plots the final results of the execute function.
"""
import os
currentFile = 'plotting'
dir_path = os.path.dirname(os.path.realpath(currentFile))
os.chdir(dir_path)
print(dir_path)
import numpy as np
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt

company_list = ['Amazon', 'Bank of New York', 'Barclays', 'Coca Cola', 'lowe‘s Companies',
                'Nike', 'Norfolk Southern Railway', 'State Street Corporation', 'Walmart']
xy = pd.read_csv('xy.csv')

V_optimization = pd.read_csv('V_optimization.csv')
V_optimization.columns = ['index', 'V']

slopes = pd.read_csv('slopes.csv')
slopes.columns = ['index', 'slopes']

htest = pd.read_csv('htest.csv')
htest.columns = ['index', 'Bank'] + company_list
htest = htest.melt(id_vars=['index'], var_name='stock')

finalwealth = pd.read_csv('finalwealthar.csv')
finalwealth.columns = ['index', 'finalwealth']

bp = pd.read_csv('bp.csv')
bp.columns = ['index', 'breakpoints']


sns.factorplot(x='index', y='value', hue='stock', data=htest, scale=2)
plt.show()

sns.set(style="darkgrid")
paper_rc = {'lines.linewidth': 1}
sns.set_context("paper", rc=paper_rc)

plt.close()
fig, ax = plt.subplots(dpi=300)
fig.set_size_inches(11.7, 8.27)
sns.factorplot(ax=ax, data=htest, x='index', y='value', hue='stock')
plt.setp(ax.collections, sizes=[0])
ax.set_xlabel('')
plt.savefig("htest.pdf")
plt.show()

File: plotting.R
rm(list = ls())
setwd('/home/laurits/Desktop/BGSE/Term 2/Stochastic Models and Optimization/Final Project/SMO')

library(ggplot2)
library(ggthemes)
library(reshape2)
library(dplyr)

htest = read.csv('htest.csv')
company_list = c('Amazon', 'Bank of New York', 'Barclays', 'Coca Cola', 'lowe‘s Companies',
                'Nike', 'Norfolk Southern Railway', 'State Street Corporation', 'Walmart')
colnames(htest) = c('index', 'Riskless asset', company_list)
wealth_test <- htest
wealth <- rowSums(wealth_test[c('Riskless asset', company_list)])

sp500_weekly <- read.csv('sp500weekly.csv')
sp500_weekly$returns <- 1 + (sp500_weekly$Close - lag(sp500_weekly$Close)) / sp500_weekly$Close
sp500_weekly <- sp500_weekly[(nrow(sp500_weekly)-156) : nrow(sp500_weekly),]
returns <- sp500_weekly$returns
dates <- as.Date(as.character(sp500_weekly$Date),'%Y-%m-%d')

wealth_sp500 <- rep(1000, length(returns))

for (i in (1 : (length(returns)-1))) {
  wealth_sp500[i+1] <- wealth_sp500[i] * returns[i]
}

diff <- wealth - wealth_sp500
sp500_comparison <- data.frame(dates, diff)
sp500_comparison <- sp500_comparison[1:(nrow(sp500_comparison)-1), ]
ggplot(sp500_comparison, aes(x = dates, diff)) + geom_line(color = 'navy', size = 1) +
  xlab('Difference') + ylab('Date') +
  ggtitle('Difference between portfolio value and S&P500') +
  geom_hline(yintercept = 0, linetype = 'dashed')
ggsave('plots/sp500_comparison_gamma0.9.pdf', width = 16, height = 9)


htest = melt(htest, id = 'index')
htest <- htest[!(htest$value == 0),]
ggplot(htest, aes(x = index, y = value, colour = variable)) + 
  geom_point(size = 5) + geom_line(size = 0.3) +
  ggtitle('Investments in different stocks over three years of the test set') +
  xlab('Period') + ylab('Investment in stocks in USD')
ggsave('plots/htest_gamma0.9.pdf', width = 16, height = 9)



finalwealth <- read.csv('finalwealthar.csv')
colnames(finalwealth) <- c('index', 'wealth')
ggplot(finalwealth, aes(x = index, y = wealth)) + 
  geom_line(color = 'navy', size = 0.5) + geom_smooth(color = 'red') +
  ggtitle('Final wealth of the training process') +
  xlab('Period') + ylab('Wealth in USD')
ggsave('plots/finalwealth_gamma0.9.pdf', width = 16, height = 9)


finaldf <- read.csv('finaldf.csv')


